internal/pkg/term/signal/signal.go

```diff
@@ -0,0 +1,36 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package signal provides a way to capture and handle operating system signals.
+package signal
+
+import (
+	"os"
+	"os/signal"
+)
+
+// Signal represents a signal handler for capturing and handling operating system signals.
+type Signal struct {
```

Comment:
I am still not convinced that we need this package. It looks pretty straightforward to me if we just use `os.Signal` directly.

Reply:
I will just remove this package and just use `os.Signal`.